plotly/tools.py

```diff
@@ -2871,3 +2934,188 @@ def get_candle_decrease(self):
         return (decrease_x, decrease_close, decrease_dif,
                 stick_decrease_y, stick_decrease_x)
 
+class _Dendrogram(FigureFactory):
+    
+    """
+    Refer to FigureFactory.create_dendrogram() for docstring.
+    """
+
+    def __init__(self, X, orientation='bottom', labels=None, colorscale=None, \
+                 width="100%", height="100%", xaxis='xaxis', yaxis='yaxis' ):
+        self.orientation = orientation
+        self.labels = labels
+        self.xaxis = xaxis
+        self.yaxis = yaxis
+        self.data = []
+        self.leaves = []
+        self.sign = {self.xaxis: 1, self.yaxis: 1}
+        self.layout = {self.xaxis: {}, self.yaxis: {}}
```

Comment:
@theengineear I am actually a bit confused about this one, but I think  I fixed it in this ebff53e5c2367b42679f2a4ddb49018ab9b93e2e commit and cleaned up here 16c74c1da67c4301e96c4a5ea0612ca405f1c7fe